lib/config/definitions.js

```diff
@@ -1491,6 +1491,15 @@ const options = [
     mergeable: true,
     cli: false,
   },
+  {
+    name: 'repositories',
```

Comment:
We definitely can't use this name as it will be too confusing to users. Can you describe more about when this is needed for Composers? And also check that we can't reuse `registryUrls` in this case?

Reply:
> We definitely can't use this name as it will be too confusing to users.

Could rename it to composerRegistries and place it in config root?

> Can you describe more about when this is needed for Composers?

Now this might be very specific for our needs, but i can see other people using this:
We have a lot of projects that use a bunch of internal composer packages that are located in our private gitlab instance. However, instead of adding the packages to each composer.json file, we created a composer plugin that injects the repositories when running `composer install/update`. Renovate extracts package info by parsing `composer.json`, thus not getting any possible required info provided by plugins. A config would allow users to inject this info themselves, e.g. this is my way of using it locally:

```
// config.js
// composer-base is provided via git in dockerfile
const rawRepos = fs.readFileSync('/usr/src/composer-base/res/repositories.json');
config.composer.repositories = JSON.parse(rawRepos);
```
> And also check that we can't reuse `registryUrls` in this case?

This would also be a possibility, but might be confusing, as a repository is a single package and would be semantically incorrect imo.

Reply:
I just realized that source urls are located in the composer.lock file ([example](https://github.com/composer/composer/blob/master/composer.lock#L12)), so it would actually be possible to extract source urls from there.

This would allow to have direct sourceUrls for all packages. The only situation where this might cause issues is if a project changes it's sourceUrl (e.g. new maintainer?), and thus not being able to find new releases anymore.

What's your opinion on this take? (reading sourceUrls from lockfile)

Reply:
I think reading sourceUrls from lockfile sounds like a good idea. In the rare case you mention, someone would need to "fix up" the master branch anyway, even if Renovate wasn't in use

Reply:
I have reworked the code to remove additional config settings in favor of reading sources from composer.lock

We probably should add test cases for composer.lock handling.